src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts

```diff
@@ -200,6 +206,18 @@ export class CheckboxSelectColumn<T extends Slick.SlickData> implements Slick.Pl
 		this._handler.unsubscribeAll();
 	}
 
+	// This call is to handle reactive changes in check box UI
+	// This DOES NOT fire UI change Events
+	public reactiveCheckboxCheck(row: number, value: boolean) {
```

Comment:
what does this get used by?

Reply:
I am bringing back the method that lets the extensions to change the cell data. Schema compare uses this method to manually check an unchecked checkbox when the database object cannot be excluded. 


Ideally, we should use the disabled flag to disable those checkboxes so the users cannot uncheck them in the first place. However, that will require some changes in the extension.

Reply:
ah ok makes sense. I forgot I was the one who added this in the past :D

Reply:
It would be great to fix https://github.com/Microsoft/azuredatastudio/blob/02697105f559d6326570ae5e2c9dab7fba24e0aa/src/sql/workbench/browser/modelComponents/table.component.ts- using an array for _checkboxColumns doesn't even make sense given that we're indexing on the column names (it should be an object/map instead)

Then this would have actually shown as a compile error when the function was removed https://github.com/Microsoft/azuredatastudio/blob/02697105f559d6326570ae5e2c9dab7fba24e0aa/src/sql/workbench/browser/modelComponents/table.component.ts#L423